libs/providers/in-memory/src/lib/in-memory-provider.ts

```diff
@@ -32,51 +34,70 @@ export class InMemoryProvider implements Provider {
     this.events.emit(ProviderEvents.ConfigurationChanged, { flagsChanged });
   }
 
-  async resolveBooleanEvaluation(flagKey: string): Promise<ResolutionDetails<boolean>> {
-    const flagValue = this.lookupFlagValueOrThrow(flagKey);
+  resolveBooleanEvaluation(flagKey: string, defaultValue: boolean, context?: EvaluationContext, logger?: Logger): Promise<ResolutionDetails<boolean>> {
+    return Promise.resolve(this.resolveFlagWithReason<boolean>(flagKey, defaultValue, context, logger));
+  }
 
-    if (typeof flagValue !== 'boolean') {
-      throw new TypeMismatchError();
-    }
+  resolveNumberEvaluation(flagKey: string, defaultValue: number, context?: EvaluationContext, logger?: Logger): Promise<ResolutionDetails<number>> {
+    return Promise.resolve(this.resolveFlagWithReason<number>(flagKey, defaultValue, context, logger));
+  }
 
-    return staticResolution(flagValue);
+  async resolveStringEvaluation(flagKey: string, defaultValue: string, context?: EvaluationContext, logger?: Logger): Promise<ResolutionDetails<string>> {
+    return Promise.resolve(this.resolveFlagWithReason<string>(flagKey, defaultValue, context, logger));
   }
 
-  async resolveStringEvaluation(flagKey: string): Promise<ResolutionDetails<string>> {
-    const flagValue = this.lookupFlagValueOrThrow(flagKey);
+  async resolveObjectEvaluation(flagKey: string, defaultValue: JsonValue, context?: EvaluationContext, logger?: Logger): Promise<ResolutionDetails<any>> {
+    return Promise.resolve(this.resolveFlagWithReason<JsonValue>(flagKey, defaultValue, context, logger));
+  }
 
-    if (typeof flagValue !== 'string') {
-      throw new TypeMismatchError();
+  private resolveFlagWithReason<T extends JsonValue | FlagValueType>(flagKey: string, defaultValue: T, ctx?: EvaluationContext, logger?: unknown): ResolutionDetails<any>{
+    try {
+      const resolutionResult = this.lookupFlagValueOrThrow(flagKey, ctx, logger)
+
+      if (typeof resolutionResult?.value != typeof defaultValue) {
+        throw new TypeMismatchError();
+      }
+
+      return resolutionResult;
+    } catch (error: unknown) {
+      return {
+        value: defaultValue, reason: error instanceof DisabledError ?
+          StandardResolutionReasons.DISABLED
+          : StandardResolutionReasons.DEFAULT
+      }
     }
-
-    return staticResolution(flagValue);
   }
 
-  async resolveNumberEvaluation(flagKey: string): Promise<ResolutionDetails<number>> {
-    const flagValue = this.lookupFlagValueOrThrow(flagKey);
-
-    if (typeof flagValue !== 'number') {
-      throw new TypeMismatchError();
+  private lookupFlagValueOrThrow(flagKey: string, ctx?: EvaluationContext, logger?: unknown): ResolutionDetails<any>{
```

Comment:
```suggestion
  private lookupFlagValue(flagKey: string, ctx?: EvaluationContext, logger?: unknown): ResolutionDetails<any>{
```

I know you didn't add this method, but I think `orThrow` is a bit gratuitous here. I think it's something that need not be in the method name.